core/cap-tbd.md

```diff
@@ -0,0 +1,209 @@
+Protocol version: TBD
+```
+
+## Simple Summary
+Add support for the BLS12-381 elliptical curve to enable zero knowledge proof systems.
+
+## Working Group
+
+This section describes the composition of the working group.
+
+## Motivation
+BLS12-381 host functions allow for cryptographic operations in the Soroban execution environment that are otherwise prohibitively expensive. It enables a variety of applications including confidentiality preserving systems, BLS signature schemes, and multiple types of scaling solutions. Pairing-friendly elliptic curves are needed for such systems and BLS12-381 is the state of the art in secure elliptical curves. 
+
+
+## Abstract
+Add host functions that enable arithmetic, pairing, and mapping operations for points on the BLS12-381 curve. 
+
+## Specification
+
+### XDR changes
+
+TBD
+
+### Semantics
+Eleven host functions are added to enable the BLS12-381 curve, which is fully defined in the [supporting docs](https://github.com/namankumar/stellar-protocol/blob/master/contents/cap-tbd/definitions_and_implementation_notes.md). G1 and G2 are two sets of points on the curve. The mathematical relationship between G1 and G2 enables advanced cryptographic systems.
+
+#### Encoding Rules
+
+Inputs to the specificed functions need to be encoded as specified.
+
+#### Encoding Rules: Field elements
+Base field element (Fp) is encoded as 64 bytes by performing BigEndian encoding of the corresponding (unsigned) integer (top 16 bytes are always zeroes). 64 bytes are chosen to have 32 byte aligned ABI (representable as e.g. bytes32[2] or uint256[2]). Corresponding integer must be less than field modulus.
+For elements of the quadratic extension field (Fp2) encoding is byte concatenation of individual encoding of the coefficients totaling in 128 bytes for a total encoding. 
+
+
+#### Encoding Rules: Points in G1 and G2
+Points in either G1 (in base field) or in G2 (in extension field) are encoded as byte concatenation of encodings of the x and y affine coordinates. Total encoding length for G1 point is thus 128 bytes and for G2 point is 256 bytes.
+
+#### Encoding Rules: Scalars for Multiplication Pperations
+Scalar for multiplication operation is encoded as 32 bytes by performing BigEndian encoding of the corresponding (unsigned) integer. Corresponding integer is not required to be less than or equal than main subgroup size.
+
+
+### Data Types
+Following types are used throughout this proposal.
+
+```
+ const BLST_G1_POINT: usize = 96; (rename after adding decoding functions)
+ const BLST_G2_POINT: usize = 192;
+ const BLST_SCALAR_SIZE: usize = 255;
+```
+### Curve Operations
+
+#### Addition in G1
+
+```rust
+fn bls12_g1add( e: &Host, p0: BLST_G1_POINT, p1: BLST_G1_POINT) -> Result <BLST_G1_POINT, Error>;
+```
+Perform addition of two points in G1. 
+Inputs should be 96 bytes each. 
+An error will be thrown if neither input is in G1, input has invalid length, or encoding rules are violated. 
```

Comment:
We don't return any errors actually - the host functions just trap in case of an error.